generator.go

```diff
@@ -0,0 +1,71 @@
+//
+//  Author:: Salim Afiune <afiune@chef.io>
+//
+
+package main
+
+// This file will have the functions that will generate random data,
+// it involve creating fake Chef Nodes and Chef Runs that can be sent
+// to the data-collector endpoint
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/go-chef/chef"
+)
+
+func generateRandomData(nodeClient chef.Client, ohaiJSON, convergeJSON, complianceJSON map[string]interface{}) (err error) {
+	channels := make([]<-chan error, config.NumNodes)
+
+	for i := 0; i < config.NumNodes; i++ {
+		nodeName := config.NodeNamePrefix + "-" + strconv.Itoa(i+1)
+		fmt.Printf(".")
+		channels[i] = ccr(nodeClient, nodeName, true, ohaiJSON, convergeJSON, complianceJSON)
+	}
+
+	for n := range merge(channels...) {
+		if n != nil {
+			fmt.Println("Error: ", n)
+			err = n
+		}
+	}
+	return err
+}
+
+func ccr(nodeClient chef.Client, nodeName string, firstRun bool,
+	ohaiJSON, convergeJSON, complianceJSON map[string]interface{}) <-chan error {
+	out := make(chan error)
+	go func() {
+		chefClientRun(nodeClient, nodeName, true, ohaiJSON, convergeJSON, complianceJSON)
+		close(out)
+	}()
+	return out
+}
+
+func merge(cs ...<-chan error) <-chan error {
```

Comment:
Still wrapping my head around these merge things, though I know I've seen this pattern before.

Reply:
The ingest service uses merge after publishing to es concurrently.

Reply:
Still, this is a bit too intense, I think we should have a `maxRequestInParallel` variable or something to control how many goroutines we want to spawn, think about it as a batch of requests. For 50,000 nodes we don't want to start 50,000 goroutines and merge them all? Maybe we do batches of 50? 

Just a thought...